src/pages/movies/index.tsx

```diff
@@ -1,91 +1,77 @@
 import {NextPage, GetServerSideProps} from "next"
 import MovieList from "../../components/MovieList"
 import PreviousPageButton from "../../components/PreviousPageButton"
+import FilterSelect from "../../components/FilterSelect"
 import {Movie} from "../../types"
 import classes from "./index.module.css"
+import { useState } from "react"
+import {useRouter} from "next/router"
+import {CustomButton} from "../../components/CustomMUI/CustomUI"
 
 
-//Hämtar data via API,
 
-/* export const getServerSideProps: GetServerSideProps = async (context) => {
-    const res = await fetch(`http://localhost:3000/api/movies`)
+export const getServerSideProps: GetServerSideProps = async (context) => {
+    const sort = context.query.sort
+    const size = context.query.size
+
+    const filters = []
+    if (sort) {
+        filters.push(`sort=${sort}`)
+    }
+    if (size) { 
+        filters.push(`size=${size}`)
+    }
+    const protocol = context.req.headers["x-forwarded-proto"] || "http";
+    const baseUrl = context.req ? `${protocol}://${context.req.headers.host}` : "";
+    const res = await fetch(`${baseUrl}/api/movies?${filters.join('&')}`)
     const data = await res.json()
-    
+
     return {
         props: {
             movies: data
         }
     }
-  }  */
+  } 
 
-export const Data = [
-    {
-        id: "1",
-        title: "Poch1",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "2",
-        title: "Poch2",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "3",
-        title: "Poch3",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "4",
-        title: "Poch4",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "5",
-        title: "Poch5",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },{
-        id: "6",
-        title: "Poch6",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    }
-    ,{
-        id: "7",
-        title: "Poch7",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },
-    {
-        id: "8",
-        title: "Poch8",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    },{
-        id: "9",
-        title: "Poch9",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
-    }
-    ,{
-        id: "10",
-        title: "Poch10",
-        thumbnail: "https://i.imgur.com/hSTzwTj.jpg"
+const Movies: NextPage<{movies: Movie[]}> = ({movies}) => {
+    const [filter, setFilter] = useState({
+        sort: "",
+        size: 9
+    })
+
+    const router = useRouter()
+
+    const handleChange = (value: string) => {
+        setFilter({
+            sort: value,
+            size: 9,
```

Comment:
By always setting the value of `size` to 9 here (and elsewhere), you will reset it whenever you change the sort. So even if you have loaded more movies, changing the sort will go back to 9. Maybe that's the desired behavior, but I would have personally preferred keeping these two states (`sort` and `size`) separate so that the user can change one without resetting the other.

If you really want to use a single object to contain both values, you could instead do `setFilter({ sort: value, size: filter.size })` or `setFilter({ ...filter, sort: value })`.